### PR TITLE
Feature/trsv2-350/implement datapush to data layer dj html

### PR DESCRIPTION
**Pull request by osimuka**
* update the google analytics data layer integration code

### CODE REVIEW

File: trade_remedies_public/templates/password/password_reset_requested.html [hunk 1]
```diff
@@ -1,13 +1,10 @@
...
-            'event': 'reset_password',
-            'user_id': '{{user.id}}'
-        });
-    </script>
-{% endif %}
+<script>
+    window.dataLayer = window.dataLayer || [];
+    window.dataLayer.push({
+        'event': 'reset_password',
+    });
```
Comment (nahmed99) [inline trade_remedies_public/templates/password/password_reset_requested.html:6]: You don't need to pass userid for this one?

File: trade_remedies_public/templates/v2/invite/sent.html [hunk 2]
```diff
@@ -3,6 +3,21 @@
...
+        'event': 'invite_team_member',
+        'user_id': '{{user.id}}',
+    });
+    {% else %}
+    window.dataLayer.push({
+        'event': 'invite_representative',
+        'user_id': '{{user.id}}',
+    });
+    {% endif %}
+</script>
```
Comment (nahmed99) [inline trade_remedies_public/templates/v2/invite/sent.html:19]: user_id (without request prefix) works here?

File: trade_remedies_public/templates/password/password_reset_requested.html (hunk 1 above)
Comment (yusufsheiqh) [reply trade_remedies_public/templates/password/password_reset_requested.html:6]: A user wouldn't be logged in when requesting a password reset.

File: trade_remedies_public/templates/v2/invite/sent.html [hunk 3]
```diff
@@ -3,6 +3,21 @@
...
+    window.dataLayer = window.dataLayer || [];
+    {% if invitation.invitation_type == 1 %}
+    window.dataLayer.push({
+        'event': 'invite_team_member',
+        'user_id': '{{user.id}}',
+    });
+    {% else %}
+    window.dataLayer.push({
+        'event': 'invite_representative',
+        'user_id': '{{user.id}}',
```
Comment (yusufsheiqh) [inline trade_remedies_public/templates/v2/invite/sent.html:16]:
```suggestion
        'user_id': '{{ request.user.id }}',
    });
    {% else %}
    window.dataLayer.push({
        'event': 'invite_representative',
        'user_id': '{{ request.user.id }}',
```

File: trade_remedies_public/templates/v2/invite/sent.html (hunk 2 above)
Comment (yusufsheiqh) [reply trade_remedies_public/templates/v2/invite/sent.html:19]: Probably not.

File: trade_remedies_public/templates/password/password_reset_requested.html (hunk 1 above)
Comment (nahmed99) [reply trade_remedies_public/templates/password/password_reset_requested.html:6]: Sounds right - misread the code (the old code was checking is user was authenticated).

File: trade_remedies_public/cases/views.py [hunk 4]
```diff
@@ -881,7 +880,7 @@ def get(
         case = self._client.get_case(case_id=case_id, organisation_id=organisation_id)
         template_grp = submission["type"]["key"]
         public_url_str = ("_" + public_str) if public_str else ""
-        template_name = f"cases/submissions/{template_grp}/upload{public_url_str}.html"
+        template_name = f"cases/submissions/application/upload{public_url_str}.html"
```
Comment (yusufsheiqh) [inline trade_remedies_public/cases/views.py:883]: Is this always going be `application`?

Comment (osimuka) [reply trade_remedies_public/cases/views.py:883]: This is actually a fix to a sentry error, I'll remove this from the PR

**Review by nahmed99 — APPROVED** [review]

**Review by yusufsheiqh — APPROVED** [review]

**Pull request merged**